cachi2/_compat/scm.py

```diff
@@ -1,196 +1,63 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 import logging
-import os
-import subprocess  # nosec
 import tarfile
 import tempfile
-import zlib
-from abc import ABC, abstractmethod
 from pathlib import Path
 
 import git
 
-from cachi2._compat.errors import (
-    FileAccessError,
-    InvalidRequestData,
-    RepositoryAccessError,
-    SubprocessCallError,
-)
-from cachi2.core.utils import get_repo_name, run_cmd, safe_extract
+from cachi2._compat.errors import InvalidRequestData, RepositoryAccessError
 
 log = logging.getLogger(__name__)
 
 
-class SCM(ABC):
-    """The base class for interacting with source control."""
+def clone_as_tarball(url: str, ref: str, to_path: Path) -> None:
+    """Clone a git repository, check out the specified revision and create a compressed tarball.
```

Comment:
I do still have to make scm.py a real module and copy its unit tests, so let me know if you are OK with keeping only this one function and dropping all the extra complexity that was relevant in Cachito

Reply:
I would start by copying only what's necessary.

Reply:
Ack, keeping only the clone function